connectors/bigtable/flink-connector-gcp-bigtable/src/main/java/com/google/flink/connector/gcp/bigtable/writer/BigtableFlushableWriter.java

```diff
@@ -33,26 +51,82 @@
  * <p>At closing, all records are flushed and clients are closed.
  */
 public class BigtableFlushableWriter {
+    String table;
     BigtableDataClient client;
     Batcher<RowMutationEntry, Void> batcher;
 
-    Sink.InitContext context;
+    List<ApiFuture<Void>> batchFutures = new ArrayList<>();
+
+    private Long totalRecordsBuffer = 0L;
+    private Long totalBytesBuffer = 0L;
+    private Counter numRecordsOutCounter;
+    private Counter numBytesOutCounter;
+    private Counter numOutEntryFailuresCounter;
+    private Counter numBatchFailuresCounter;
+    private Histogram numEntriesPerFlush;
+
+    private static final Logger logger = LoggerFactory.getLogger(BigtableFlushableWriter.class);
+
+    private static final Integer BATCHER_CLOSE_TIMEOUT_SECONDS = 60;
 
     public BigtableFlushableWriter(
-            BigtableDataClient client, Sink.InitContext sinkInitContext, String table) {
+            BigtableDataClient client, WriterInitContext sinkInitContext, String table) {
         this.client = client;
+        this.table = table;
         this.batcher = client.newBulkMutationBatcher(TableId.of(table));
-        this.context = sinkInitContext;
+
+        // Instantiate Metrics
+        this.numRecordsOutCounter =
+                sinkInitContext.metricGroup().getIOMetricGroup().getNumRecordsOutCounter();
```

Comment:
can you add some null protections here? ie, what if sinkinitContext is null? of the IOMetricGroup?

Reply:
I had them before with ifs but ended up removing since I feel they will be hiding other issues. 

Checking other connectors like [Kafka](https://github.com/apache/flink-connector-kafka/blob/main/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaWriter.java#L152) 

I will add some `checkNoNull` so that the initialization fails if no context

Reply:
Added the checkNotNull for the context,metricGroup cannot be null